benchmark/nrt_dataset.py

```
@@ -55,19 +51,21 @@ def write(cls, data: tuple[dict, JointNestedRaggedTensorDict], data_dir: Path):
-        if self.max_seq_len is not None:
-            L = len(dynamic_data)
-            if L > self.max_seq_len:
-                start = np.random.randint(0, L - self.max_seq_len)
-                dynamic_data = dynamic_data[start : start + self.max_seq_len]
+    @TimeableMixin.TimeAs
+    def _load_dynamic_data(self, i, start=None, end=None) -> JointNestedRaggedTensorDict:
+        if self.max_seq_len is None:
+            return self.dynamic_data[i]
+
+        L = end - start
+        if L <= self.max_seq_len:
```

Ruff

Yoda condition detected

Rewrite as `self.max_seq_len >= L`

(SIM300)